pyccel/parser/semantic.py

```diff
@@ -1709,6 +1709,13 @@ def _ensure_inferred_type_matches_existing(self, class_type, d_var, var, is_auga
                 tmp_result = PyccelAdd(var, rhs)
                 result_type = tmp_result.class_type
                 raise_error = var.class_type != result_type
+            elif isinstance(var.class_type, InhomogeneousTupleType) and \
+                    isinstance(class_type, HomogeneousTupleType):
+                raise_error = any(a != class_type.element_type for a in var.class_type) or var.shape[0] != d_var['shape'][0]
```

Comment:
@jalalium these lines are the source of the errors you are seeing. Do you have a suggestion for how I should handle this? I can make a tuple with 1 element inhomogeneous by default. Usually homogenous is easier to match with everything else but not here because we also need to check zhe size. Maybe I need a recursive function